src/Connection.php

```diff
@@ -212,21 +213,31 @@ public function scrollSelect($params, $bindings = [])
 
         $numResults = count($results['hits']['hits']);
 
-        if ( $limit >= $numResults ){
-            $results['scrollCursor'] = $this->scroll($scrollId, $scrollTimeout, $limit - $numResults);
+        foreach ($results['hits']['hits'] as $result) {
+            yield $result;
         }
 
-        return $results;
+        if ( $limit >= $numResults ){
+            foreach ($this->scroll($scrollId, $scrollTimeout, $limit - $numResults) as $result) {
+                yield $result;
+            }
+        }
     }
 
     /**
      * Run a select statement against the database using an Elasticsearch scroll cursor.
+     *
+     * @param  string $scrollId
+     * @param  string $scrollTimeout
+     * @param  int $limit
+     * @return \Generator
      */
-    public function scroll($scrollId, $scrollTimeout, $limit){
+    public function scroll(string $scrollId, string $scrollTimeout = '30s', int $limit = null)
```

Comment:
If `$limit` is an integer could it's default value be 0? You might then be able to remove the `!$limit` part of the while loop on line 240.